src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -162,6 +204,9 @@ PG_FUNCTION_INFO_V1(citus_shard_cost_by_disk_size);
 PG_FUNCTION_INFO_V1(citus_validate_rebalance_strategy_functions);
 PG_FUNCTION_INFO_V1(pg_dist_rebalance_strategy_enterprise_check);
 
+#define MAX_LOGGED_IGNORED_MOVES 5
+#define MAX_LOGGED_IGNORED_MOVES_STRING "5"
```

Comment:
wonder if we want to make this a hidden GUC, might be easier for runtime debugging when we want to see more lines. With something like `-1` to just print all. I don't know if that is ever required but could help in diving deep on reported problems with the balancing choosing strange moves.